scs2-simulation/src/main/java/us/ihmc/scs2/simulation/physicsEngine/contactPointBased/ContactPointBasedRobotPhysics.java

```diff
@@ -122,7 +153,11 @@ public void doForwardDynamics(Vector3DReadOnly gravity)
                                                        }
                                                        return simJoint.isPinned() ? JointSourceMode.ACCELERATION_SOURCE : JointSourceMode.EFFORT_SOURCE;
                                                     });
-      forwardDynamicsCalculator.compute();
+      getJointTauFromControllers();
+      // NOTE: by the time this method is called, the joint torques from simulation efforts will have been called, and jointsTauSimulationEffects
+      // will have been updated. Therefore, the contributions can be summed.
+      sumJointTauContributions();
+      forwardDynamicsCalculator.compute(jointsTau);
```

Comment:
This might be poor. In `ContactBasedPhysicsEngine.simulate()`, we call the above `computeJointSimulationEffects` before this overall method, so it's guaranteed that `jointsTauSimulationEffects` is populated. You wouldn't know that though, without this comment.

Reply:
I suppose we could add a javadoc to the `doForwardDynamics` method to mention that joint torques are expected to be all computed.